.github/workflows/check-main-status.yml

```diff
@@ -0,0 +1,63 @@
+name: Check Main Branch Status
+on:
+  pull_request:
```

Comment:
Should not run on PRs, otherwise you cannot fix a broken main with a PR.

Reply:
The PR is under work. Will skip the check when a certain label is on the PR.